### ISSUE TITLE
Flag `-XepAllDisabledChecksAsWarnings` doesn't quite do what its name implies

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#486]
Passing `-XepAllDisabledChecksAsWarnings` sets `ErrorProneOptions.isEnableAllChecks()`, which [leads to](https://github.com/google/error-prone/blob/de639a1ef4d962b984741ed346f0519c85eb6122/check_api/src/main/java/com/google/errorprone/scanner/ScannerSupplier.java#L163):

```java
    if (errorProneOptions.isEnableAllChecks()) {
      disabled.forEach(c -> severities.put(c, checks.get(c).defaultSeverity()));
      disabled.clear();
    }
```

As you can see, all disabled checks are indeed enabled, but with their _default_ severity. This means that suggestions enabled through this flag will not cause a build failure when compiling with `-Werror`. That's unintuitive given the flag's name.

NB: this flag does not appear to be documented on the [website](http://errorprone.info/docs/flags).

**Comment by Stephan202** [comment]
My personal preference would be for this functionality to be split up in two flags:

* `-XepEnableAllChecks`. Does what `-XepAllDisabledChecksAsWarnings` does currently.
* `-XepAllSuggestionsAsWarnings`. Analogous to `-XepAllErrorsAsWarnings`, but in the opposite direction. Affects all suggestions, irrespective of whether they are enabled by default. (Currently no suggestions seem enabled by default,  but I suppose that may change, and severity and "default enabledness" seem orthogonal concerns.)

**Comment by eaftan** [comment]
This flag is part of some in-progress work to enable a "--fix" mode.  We'll add documentation when we're ready to announce it.

@nglorioso, can you take a look?
